scrapy/pipelines/images.py

```diff
@@ -115,7 +118,16 @@ def file_downloaded(self, response, request, info, *, item=None):
 
     def image_downloaded(self, response, request, info, *, item=None):
         checksum = None
-        for path, image, buf in self.get_images(response, request, info, item=item):
+        image_stream = self.get_images(response, request, info, item=item)
+        while True:
+            try:
+                path, image, buf = next(image_stream)
+            except OSError:
+                logger.exception('Could not process image')
+                continue
+            except Exception:
+                logger.exception('Stopped processing images')
+                break
```

Comment:
The loop should not `break` because one image failed. It should continue processing the other images.

Reply:
Do we expect exceptions other than `OSError` happening? If we add `except StopIteration`, maybe we don’t need this `except` here at all.

Reply:
So, I should replace `break `with `continue ` or delete it at all, as long as I add `except StopIteration` ?

Reply:
I guess we could leave `Exception` for subclasses raising exceptions we have no control over. So `continue` here and `except StopIteration` before this `except`, I think.